addons/stock/models/stock_move.py

```diff
@@ -493,6 +492,18 @@ def _set_lot_ids(self):
                     move_lines_commands.append((0, 0, move_line_vals))
             move.write({'move_line_ids': move_lines_commands})
 
+    @api.depends('picking_id', 'picking_type_id', 'date')
+    def _compute_reservation_date(self):
+        for move in self:
+            if move.state in ['draft', 'confirmed', 'waiting', 'partially_available']:
```

Comment:
you may have error if you do not give any value to the field (for instance cancelled or done moves)